File: scripts/sup.pl
#!/usr/bin/env perl
BEGIN { unshift @INC, "../lib"; }
use Data::Dumper;
use JSON;
use LWP::UserAgent;
use Net::MQTT::Simple::SSL;

my $mqtt_host='mqtt.hq.thebikeshed.io';
my $tlsopts= {
               SSL_ca_file       => '/etc/ssl/certs/ca.crt',
               SSL_cert_file     => '/etc/ssl/certs/localhost.crt',
               SSL_key_file      => '/etc/ssl/private/localhost.ckey',
               SSL_verifycn_name => $mqtt_host,
             };

my $mqtt = Net::MQTT::Simple::SSL->new( $mqtt_host, $tlsopts );

sub mqttsay{
  my ($where, $who, $what) = @_;
  print "$where <$who>: $what\n";
  my $mqtt_response = Net::MQTT::Simple::SSL->new( $respond_host, $tlsopts);
  $mqtt_response->publish("irc/room/$where/nick/crunchy/say" => "$what");
}

sub callbacks {
  $mqtt->run( "irc/room/+/nick/+/said" => sub {
                                                my ($topic, $message) = @_;
                                                if( $topic =~ /irc\/room\/(.*)\/nick\/(.*)\/said/){
                                                  my ($where, $who) = ($1, $2);
                                                  my $what = $message;
                                                  actions($where, $who, $what);
                                                }
                                               }
            );
}

sub actions {
  my ($where, $who, $what) = @_;
  for ( $what ) {
    /^sup?/ && do { mqttsay($where, $who, "not much."); last; };
  }
}

callbacks

File: scripts/sup.py
import paho.mqtt.client as mqtt
import ssl
import re

client = mqtt.Client()

MQTT_HOST = 'localhost'

client.tls_set('/etc/ssl/private/ca.crt',
               certfile='/etc/ssl/certs/localhost.crt',
               keyfile='/etc/ssl/private/localhost.ckey',
               cert_reqs=ssl.CERT_REQUIRED,
               tls_version=ssl.PROTOCOL_TLSv1)


def on_message(client, userdata, message):
    if re.match(r'^sup?', message.lower()):
        client.publish('not much.')


def on_connect(client, userdata, flags, rc):
    client.subscribe("#")
    client.publish('Hello mqtt')


client.on_connect = on_connect
client.on_message = on_message

client.connect(MQTT_HOST)
